src/main/java/roomescape/global/config/WebMvcConfiguration.java

```
@@ -27,16 +32,34 @@ public WebMvcConfiguration(
     @Override
     public void addInterceptors(InterceptorRegistry registry) {
         registry.addInterceptor(checkRoleInterceptor)
-            .addPathPatterns("/admin/**");
+                .addPathPatterns("/admin/**");
 
         registry.addInterceptor(checkUserInterceptor)
-            .addPathPatterns("/reservation")
-            .addPathPatterns("/reservations")
-            .addPathPatterns("/login/check");
+                .addPathPatterns("/reservation")
+                .addPathPatterns("/reservations")
+                .addPathPatterns("/login/check");
     }
 
     @Override
     public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
         resolvers.add(argumentResolver);
     }
+
+    @Bean
+    public RestClient restClient() {
+        String TOSS_PAYMENTS_URL = "https://api.tosspayments.com/v1/payments/confirm";
+
+        return RestClient.builder()
+                .baseUrl(TOSS_PAYMENTS_URL)
+                .requestFactory(clientHttpRequestFactory())
+                .build();
+    }
+
+    private ClientHttpRequestFactory clientHttpRequestFactory() {
+        SimpleClientHttpRequestFactory factory = new SimpleClientHttpRequestFactory();
+        factory.setConnectTimeout(3000);
+        factory.setReadTimeout(3000);
```

connection time out과 read timeout을 각각 3000을 설정한 이유가 있을까요?

---

Fail Fast가 중요하다고 생각해서 timeout 범위를 최소한의 시간으로 3초로 설정했습니다.

하지만 추가로 toss payment timeout 적정 시간을 찾아보니, 어느곳에서는 10초를, 어느 문서에서는 30초를 권장한다고 나와있어서 어떤 것을 참조하면 좋을지 어려웠습니다.
그럼에도 30초는 너무 긴 시간인 것같아서 10초로 readTimeout을 변경하였습니다.

[10초 권장](https://tossdev.github.io/qna.html)
[10초 권장](https://docs.tosspayments.com/resources/glossary/timeout)